cmd/gce-pd-csi-driver/main.go

```diff
@@ -245,11 +245,15 @@ func handle() {
 		if err != nil {
 			klog.Fatalf("Failed to set up metadata service: %v", err.Error())
 		}
+		isDataCacheEnabledNodePool, err := isDataCacheEnabledNodePool(ctx, *nodeName)
+		if err != nil {
+			klog.Fatalf("Failed to get node info from API server: %v", err.Error())
```

Comment:
This implementation we will not get signal from alerts for get node info from API server? But I saw the above is doing the same for metadataservice. Is this expected? @mattcary

Reply:
I'm not sure what you mean here? err != nil if the api call fails.

Reply:
But it only logged as fatal and continued, so what if the isDataCacheEnabledNodePool is always broken?

Reply:
klog.Fatalf kills the program, no? https://pkg.go.dev/k8s.io/klog/v2#Fatalf